pkg/codec/byte_string_modifier.go

```diff
@@ -0,0 +1,307 @@
+package codec
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+// AddressModifier defines the interface for encoding, decoding, and handling addresses.
+// This interface allows for chain-specific logic to be injected into the modifier without
+// modifying the common repository, making the implementation flexible and adaptable for
+// different blockchain ecosystems (e.g., Ethereum, Solana).
+type AddressModifier interface {
+	// Converts a byte array representing an address into its string form.
+	EncodeAddress([]byte) (string, error)
+	// Converts a string representation of an address back into its byte array form.
+	DecodeAddress(string) ([]byte, error)
+	// Returns the expected byte length of the address for the specific chain.
+	Length() int
+}
+
+// NewAddressBytesToStringModifier creates and returns a new modifier that transforms address byte
+// arrays to their corresponding string representation (or vice versa) based on the provided
+// AddressModifier.
+//
+// The fields parameter specifies which fields within a struct should be modified. The AddressModifier
+// is injected into the modifier to handle chain-specific logic (e.g., different encoding schemes for
+// Ethereum or Solana addresses). This happens during the contractReader relayer configuration.
+func NewAddressBytesToStringModifier(fields []string, modifier AddressModifier) Modifier {
+	fieldMap := map[string]bool{}
+	for _, field := range fields {
+		fieldMap[field] = true
+	}
+
+	m := &bytesToStringModifier{
+		modifier: modifier,
+		modifierBase: modifierBase[bool]{
+			fields:           fieldMap,
+			onToOffChainType: map[reflect.Type]reflect.Type{},
+			offToOnChainType: map[reflect.Type]reflect.Type{},
+		},
+	}
+
+	// Modify field for input using the modifier to convert the byte array to string
+	m.modifyFieldForInput = func(_ string, field *reflect.StructField, _ string, _ bool) error {
+		t, err := convertBytesToString(field.Type, field.Name, modifier.Length())
+		if err != nil {
+			return err
+		}
+		field.Type = t
+		return nil
+	}
+
+	return m
+}
+
+type bytesToStringModifier struct {
+	// Injected modifier that contains chain-specific logic
+	modifier AddressModifier
+	modifierBase[bool]
+}
+
+func (t *bytesToStringModifier) RetypeToOffChain(onChainType reflect.Type, itemType string) (tpe reflect.Type, err error) {
+	defer func() {
+		// StructOf can panic if the fields are not valid
+		if r := recover(); r != nil {
+			tpe = nil
+			err = fmt.Errorf("%w: %v", types.ErrInvalidType, r)
+		}
+	}()
+
+	if len(t.fields) == 0 {
```

Comment:
what does this do?

Reply:
return the original type without performing any transformation, because no fields require modification

Reply:
makes sense, would it make sense to return an error here?

Reply:
It makes sense to me, I think it will help avoiding confusions. However, the [modifierBase RetypeToOffChain](https://github.com/smartcontractkit/chainlink-common/blob/8166e6555b2330ba0abfc747a6a78d0685394d89/pkg/codec/modifier_base.go#L30-L34) is not returning an error under this condition. Do you want me to change it there too?

Reply:
I think I found a way to reuse the modifier base functionality. Let me try something and I'll push some changes

Reply:
Removed the check from the new modifier as we will rely on the `modiferBase RetypeToOffChain` now. It seems better to leave that condition as it is in the `modifierBase` for this development as other modifiers start to fail. 

I can create a new ticket to check this later and see if it makes sense for all the modifiers to have an error there